### PR TITLE
docs: basic netlify configuration for book

### PR DESCRIPTION
**Pull request by salasberryfin**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

This PR adds the basic `netlify` configuration (plus minimal make tools for building the book) so that we can [request a `netlify` site](https://github.com/kubernetes/community/blob/c5243d80e565a3292f446f6f54c82e57ea7b1602/github-management/subproject-site-requests.md#requesting-a-netlify-site) and a [project subdomain](https://github.com/kubernetes/community/blob/c5243d80e565a3292f446f6f54c82e57ea7b1602/github-management/subproject-site-requests.md#requesting-a-subproject-domain).

Together with this PR, I'm opening the necessary requests in the https://github.com/kubernetes/k8s.io/ repository.

Then, when the site is up and running, I'll create a separate PR to add the base structure/contents of the book, so that we can later iterate over this and add new/improve existing topics.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

This the initial PR on an effort to sort the CAPG docs into a book https://github.com/kubernetes-sigs/cluster-api-provider-gcp/issues/490.

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```release-note
Add Netlify configuration for book
```

### CODE REVIEW

**Comment by salasberryfin** [comment]
@richardcase I think there's a typo in the to-test label and prow didn't start the checks

**Review by cpanato — COMMENTED** [review]
/ok-to-test

**Comment by salasberryfin** [comment]
Fixed tests and it is now ready to be reviewed.

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    command = "make -C docs/book build"
+    publish = "docs/book/book"
+
+[build.environment]
+    GO_VERSION = "1.21"
+
+# Standard Netlify redirects
+[[redirects]]
+    from = "https://kubernetes-sigs-cloud-provider-gcp.netlify.com/*"
+    to = "https://cloud-provider-gcp.sigs.k8s.io/:splat"
```
Comment (vincepri) [inline netlify.toml:12]:
```suggestion
    to = "https://cluster-api-gcp.sigs.k8s.io/:splat"
```

Comment (salasberryfin) [reply netlify.toml]: My mediocre copy/pasting skills

File: hack/tools/go.mod [hunk 2]
```diff
@@ -4,21 +4,26 @@ go 1.21
 
 toolchain go1.21.6
 
-require sigs.k8s.io/cluster-api/hack/tools v0.0.0-20240116064735-bfe8d0d16ff3
+require sigs.k8s.io/cluster-api/hack/tools v0.0.0-20221121093230-b1688621953c
```
Comment (vincepri) [inline hack/tools/go.mod:7]: why downgrade?

**Review by cpanato — APPROVED** [review]
/approve
/lgtm

thanks for working on this

**Pull request merged**